firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/RemoteSettings.kt

```diff
@@ -54,38 +64,102 @@ internal class RemoteSettings(val context: Context) : SettingsProvider {
+      } catch (ex: Exception) {
+        // TODO(visum) Failed to get installations ID
       }
+    }
+
+    if (installationId == null) {
+      fetchInProgress = false
+      return
+    }
+
+    val options =
+      mapOf(
+        "X-Crashlytics-Installation-ID" to installationId as String,
+        "X-Crashlytics-Device-Model" to appInfo.deviceModel,
+        // TODO(visum) Add OS version parameters
+        // "X-Crashlytics-OS-Build-Version" to "",
+        // "X-Crashlytics-OS-Display-Version" to "",
+        "X-Crashlytics-API-Client-Version" to appInfo.sessionSdkVersion
+      )
+    runBlocking {
+      configsFetcher.doConfigFetch(
+        headerOptions = options,
+        onSuccess = {
+          var sessionsEnabled: Boolean? = null
+          var sessionSamplingRate: Double? = null
+          var sessionTimeoutSeconds: Int? = null
+          var cacheDuration: Int? = null
+          if (it.has("app_quality")) {
+            val aqsSettings = it.get("app_quality") as JSONObject
+            try {
+              if (aqsSettings.has("sessions_enabled")) {
+                sessionsEnabled = aqsSettings.get("sessions_enabled") as Boolean?
+              }
 
-      // Check if cache is expired. If not, return
-      // Initiate a fetch. On successful response cache the fetched values
+              if (aqsSettings.has("sampling_rate")) {
+                sessionSamplingRate = aqsSettings.get("sampling_rate") as Double?
+              }
 
+              if (aqsSettings.has("session_timeout_seconds")) {
+                sessionTimeoutSeconds = aqsSettings.get("session_timeout_seconds") as Int?
+              }
+
+              if (aqsSettings.has("cache_duration")) {
+                cacheDuration = aqsSettings.get("cache_duration") as Int?
+              }
+            } catch (exception: JSONException) {
+              // TODO(visum) Log failure to parse the configs fetched.
+            }
+          }
+
+          sessionsEnabled?.let { settingsCache.updateSettingsEnabled(sessionsEnabled) }
+
+          sessionTimeoutSeconds?.let {
+            settingsCache.updateSessionRestartTimeout(sessionTimeoutSeconds)
+          }
+
+          sessionSamplingRate?.let { settingsCache.updateSamplingRate(sessionSamplingRate) }
+
+          cacheDuration?.let { settingsCache.updateSessionCacheDuration(cacheDuration) }
+            ?: let { settingsCache.updateSessionCacheDuration(86400) }
+
+          settingsCache.updateSessionCacheUpdatedTime(System.currentTimeMillis())
+          fetchInProgress = false
+        },
+        onFailure = {
+          // Network request failed here.
+          // TODO(visum) Log error in fetching configs
+          // Logger.logError("[Settings] Fetching newest settings failed with error: \(error)")
+          fetchInProgress = false
+        }
+      )
     }
   }
+
+  companion object {
+    private const val SESSION_CONFIGS_NAME = "firebase_session_settings"
+    private var fetchInProgress = false
```

Comment:
Should `fetchInProgress` be defined in the class? I think we will only make one instance of this class, but it might get weird in the future if this var is in the companion object.

Reply:
Done.